ccswbs/gus
# Add local menus and upgrade to Bootstrap 5

# Summary of Changes

The purpose of this PR is twofold: pull data from Drupal menus and upgrade to Bootstrap 5

## Frontend
- Remove gatsby-plugin-preact
- Restore `html.js` to Gatsby default and move external scripts and stylesheets to `gatsby-ssr.js` (best practice)
- Remove gatsby menu links plugin (no longer used or needed)
- Remove old menu code from `gatsby-node.js` and add new schema and functions
- Create HeaderMenu component
- Upgrade Breadcrumbs component
- Display unpublished pages and programs in separate section on Gatsby index page
- Upgrade js and css to Bootstrap 5
- Add Bootstrap 5 size and spacing classes to `buttons.js`
- Refactor Links Widget components for Bootstrap 5 compatibility

## Backend
- Drupal multidev is https://menus-chug.pantheonsite.io/
- Pantheon dashboard link is https://dashboard.pantheon.io/sites/2d1d74da-fc5c-454b-9952-92bf214d372d#menus
- Add [Drupal core patch for jsonapi link fields](https://www.drupal.org/project/drupal/issues/3066751)
- Enable `Include count in collection queries` (as per recommendation in https://www.npmjs.com/package/gatsby-source-drupal?activeTab=readme)
- Add view and administer menu permission to `Viewer` role
- Enable HTTP basic auth module (to let the Gatsby menu-viewing role authenticate itself)
- Add Office of the Chancellor menu

# Test Plan
- Go to https://tqtest.gatsbyjs.io/
- Click `A Place to Grow`
- Verify you see the Place to Grow menu in the page-specific menu area
- Click `Recruitment` menu link and verify the drop-down to subpages appears. Click a subpage
- Verify the Breadcrumb trail shows `Home > Recruitment > Whatever subpage you chose`
- Verify the `Home` Breadcrumb link takes you back to `A Place to Grow`
- Go back to https://tqtest.gatsbyjs.io/ and repeat the above steps, this time starting with `Role of the Chancellor` - it has no submenus, so you just need to verify all the menu links go to the correct pages

~~**Note:** The automatic Pull Request build against https://api.liveugconthub.uoguelph.dev/ will fail because the live Drupal site doesn't have the necessary permissions to pull menu data~~ Resolved with the addition of missing fields to `gatsby-node.js` schema - PR build now works: https://build-22fbac1d-e065-475a-8c52-2855f943e152.gtsb.io/ (though the page-specific menu area will be blank)
